lnutils/log.go

```diff
@@ -36,3 +38,13 @@ func NewSeparatorClosure() LogClosure {
 		return strings.Repeat("=", 80)
 	}
 }
+
+// LogPubKey returns a slog attribute for logging a public key in hex format.
+func LogPubKey(key string, pubKey *btcec.PublicKey) btclog.Attr {
+	// Handle nil pubkey gracefully, although callers should ideally prevent
+	// this.
+	if pubKey == nil {
+		return btclog.String(key, "<nil>")
+	}
+	return btclog.Hex(key, pubKey.SerializeCompressed())
```

Comment:
do we want the full pub key or should we make use of one of the helper `HexN` functions to only print a couple identifying characters?

Reply:
Went with `Hex6` here.